prototype_source/maskedtensor_adagrad.rst

```diff
@@ -0,0 +1,182 @@
+Efficiency of writing "sparse" semantics for Adagrad
+====================================================
+
+`Issue 1369 <https://github.com/pytorch/pytorch/issues/1369>`__ discussed the additional lines of code
+that were introduce while writing "sparse" semantics for Adagrad.
+But really the code doesn't use sparsity as a compression and optimization technique,
+it wants to use masked semantics. We worked around this by introducing one-off semantics and operators
```

Comment:
I'm not sure I understand the difference here--is this trying to say "But really, the code would gain better compression wins and optimizations with masked semantics, not sparsity"?

Reply:
Thanks, it is a bit confusing - it's really trying to say "the code is really using sparsity as a proxy for masked semantics when sparsity is meant to be used as a compression/optimization technique"